### PR TITLE
Adding Singleton Support

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -182,15 +195,15 @@ func (cli *Cli) Run() {
 		}
 		os.Exit(0)
 	}
-reset:
-	//Get user input
-	fmt.Print(">>>")
+	for {
```
Comment (ghost) [inline cli/cli.go:185]: Was not keen on the use of GOTO and have opted to use a 'forever' loop instead.

File: cli/cli.go [hunk 2]
```diff
@@ -20,6 +21,18 @@ type Cli struct {
 	Scanner        *readline.Instance
 }
 
+var instance *Cli
```
Comment (AlexsJones) [inline cli/cli.go]: I have mixed feelings about using singletons and global antipatterns

File: cli/cli.go (hunk 1 above)
Comment (AlexsJones) [reply cli/cli.go:185]: good change

**Comment by AlexsJones** [comment]
Happy to merge if you can let me know the reason for the singleton @RenegadeTech 
And why it's better than the current invocation.

File: cli/cli.go (hunk 2 above)
Comment (ghost) [reply cli/cli.go]: Okay, that makes sense.

Comment (ghost) [reply cli/cli.go]: I am going to remove the support for singleton and move a more functional approach which I feel is more of the idiom of GoLang.

**Comment by ghost** [comment]
The original idea for the singleton was so that other packages could wire themselves up to the global context and then main would only need to obtain the global context and then run.

That is why I wanted to add the singleton.

**Comment by AlexsJones** [comment]
Need to revisit this @RenegadeTech if you still think its worth doing we can merge, though I'm a bit on the fence.

**Review by AlexsJones — APPROVED** [review]

**Pull request closed without merging**